### PR TITLE
DDCE-5337 | Added CSOP V5.

### PR DESCRIPTION
**Pull request by slavvco**

### CODE REVIEW

File: test/services/ProcessCsvServiceSpec.scala [hunk 1]
```diff
@@ -112,8 +114,8 @@ class ProcessCsvServiceSpec
     }
 
     "process a row and return an empty list if there are no errors" in {
-      val source = convertToAkkaSource(new File(System.getProperty("user.dir") + "/test/resources/copy/CSOP_OptionsGranted_V4.csv"))
-      val dataValidator = new DataValidator(ConfigFactory.load.getConfig("ers-csop-granted-validation-config"))
+      val source = convertToAkkaSource(new File(System.getProperty("user.dir") + "/test/resources/copy/CSOP_OptionsGranted_V5.csv"))
```
Comment (MRBeresford) [inline test/services/ProcessCsvServiceSpec.scala:117]: Lots of these tests you replace V4 with V5, would it be worth "duplicating" the tests in a similar way to how you did in `test/services/DataGeneratorSpec.scala`? or given that both schemas have identical validation logic and just the row name has changed this is sufficent?

File: test/services/ParserTest.scala [hunk 2]
```diff
@@ -64,7 +64,7 @@ class ParserTest
   object TestDataParser extends DataParser {
     val logger: Logger = Logger(getClass)
   }
-  object TestDataGenerator extends DataGenerator(mockAuditEvents, mockMetrics, mockParserUtil, realErsValidationConfigs, mockErsUtil, mockErsValidator)
+  object TestDataGenerator extends DataGenerator(mockAuditEvents, mockMetrics, mockParserUtil, realErsValidationConfigs, mockErsUtil, mockErsValidator, mockAppConfig)
```
Comment (MRBeresford) [inline test/services/ParserTest.scala:67]: There are some tests on L110 -> L126 looking at CSOP sheet name mapping to the correct sheetType and sheetId, you may have already looked at this but if not might be worth adding some tests for CSOP V5 sheet names as well?

**Review by MRBeresford — COMMENTED** [review]
Looks good, just a few things to discuss

File: app/services/DataGenerator.scala [hunk 3]
```diff
@@ -161,6 +163,15 @@ class DataGenerator @Inject()(auditEvents: AuditEvents,
     }
   }
 
+  private def getValidator(sheetName: String): DataValidator = {
```
[the remark this answers is not in the document]
Comment (slavvco) [reply app/services/DataGenerator.scala:166]: yep, good spot, thanks

File: test/services/ParserTest.scala (hunk 2 above)
Comment (slavvco) [reply test/services/ParserTest.scala:67]: sure, can add more

File: test/services/ProcessCsvServiceSpec.scala (hunk 1 above)
Comment (slavvco) [reply test/services/ProcessCsvServiceSpec.scala:117]: as discussed, leaving as is

**Review by MRBeresford — APPROVED** [review]
LGTM

**Pull request merged**